packages/node_modules/@ciscospark/redux-module-media/src/reducer.js

```diff
@@ -3,142 +3,81 @@ import {fromJS} from 'immutable';
 import {
   UPDATE_STATUS,
   UPDATE_CALL_STATE,
-  UPDATE_LOCAL_MEDIA,
-  UPDATE_REMOTE_MEDIA,
-  CALL_CONNECTED,
-  CALL_DISCONNECTED,
+  CONNECT_CALL,
+  REMOVE_CALL,
   STORE_CALL,
-  CALL_INCOMING,
-  ANSWER_INCOMING_CALL,
-  REMOVE_INCOMING_CALL,
-  CHECKING_WEBRTC_SUPPORT,
+  CHECK_WEBRTC_SUPPORT,
   UPDATE_WEBRTC_SUPPORT
 } from './actions';
 
+const initialCallObject = fromJS({
+  instance: null,
+  spaceId: null,
+  callState: {},
+  startTime: null
+});
+
 export const initialState = fromJS({
-  call: null,
-  startTime: null,
-  incomingCall: {
-    hasAcknowledged: false,
-    call: null
-  },
-  callState: {
-    sendingAudio: null,
-    sendingVideo: null
-  },
+  calls: {},
```

Comment:
Order they come in really doesn't matter. In recents it's the order of last seenable activity that matters.